erts/emulator/hipe/hipe_amd64.c

```diff
@@ -197,14 +203,15 @@ void *hipe_make_native_stub(void *callee_exp, unsigned int beamArity)
     codep[0] = beamArity;
     codep += 1;
 
-    /* jmp callemu; 5 bytes */
-    callEmuOffset = (unsigned char*)nbif_callemu - (code + codeSize);
-    codep[0] = 0xe9;
-    codep[1] =  callEmuOffset        & 0xFF;
-    codep[2] = (callEmuOffset >>  8) & 0xFF;
-    codep[3] = (callEmuOffset >> 16) & 0xFF;
-    codep[4] = (callEmuOffset >> 24) & 0xFF;
-    codep += 5;
+    /* jmp callemu; 12 bytes */
```

Comment:
You should have the comment say the right instruction sequence; i.e. `movabsq callemu, %rax; jmpq *%rax`.

Reply:
I agree that using trampolines on amd64 is the proper way to handle PIEs.  Linkage should be optimized to bypass the trampolines whenever possible, so non-PIEs aren't penalized (I believe we do something like that on PowerPC).  The compiler proper shouldn't need any changes.